### PR TITLE
Document worker thread pool project settings and `TreeItem.uncollapse_tree()`

### PR DESCRIPTION
**Pull request by MewPurPur**
Adds some missing docs.

https://github.com/godotengine/godot/pull/76525#issuecomment-1545474260

### CODE REVIEW

**Comment by MewPurPur** [comment]
@RandomShaper Could you look at these descs, please?

File: doc/classes/ProjectSettings.xml [hunk 1]
```diff
@@ -2699,17 +2699,19 @@
 			[/codeblock]
 		</member>
 		<member name="threading/worker_pool/low_priority_thread_ratio" type="float" setter="" getter="" default="0.3">
+			If [member threading/worker_pool/use_system_threads_for_low_priority_tasks] is [code]false[/code], this value determines the ratio of low-priority tasks that can be executed at a time by [WorkerThreadPool]'s threads. For example, if 10 threads are available and this value is set to [code]0.3[/code], at most 3 low-priority tasks can be executed by the [WorkerThreadPool] at a time.
```
Comment (RandomShaper) [inline doc/classes/ProjectSettings.xml:2702]: Correct. You can add that the number of low-pri tasks is clamped so there are never fewer than one nor more than the total thread count minus one.

File: doc/classes/ProjectSettings.xml [hunk 2]
```diff
@@ -2699,17 +2699,19 @@
...
 		</member>
 		<member name="threading/worker_pool/max_threads" type="int" setter="" getter="" default="-1">
 			Maximum number of threads to be used by [WorkerThreadPool]. Value of [code]-1[/code] means no limit.
 		</member>
 		<member name="threading/worker_pool/use_system_threads_for_low_priority_tasks" type="bool" setter="" getter="" default="true">
+			If [code]true[/code], [WorkerThreadPool]'s threads will be reserved for high-priority tasks.
```
Comment (RandomShaper) [inline doc/classes/ProjectSettings.xml:2708]: The actual meaning is this:
If true, low priority tasks won't run on the threads from the pool. Instead, for each low priority task a new thread will be created and finally destroyed when the task is over. Ratio of low-prio tasks still apply.

File: doc/classes/ProjectSettings.xml (hunk 1 above)
Comment (MewPurPur) [reply doc/classes/ProjectSettings.xml:2702]: I don't understand the meaning of this, is this saying that if `use_system_threads_for_low_priority_tasks`, then there is at least one low-priority task that can be processed by worker threads at a time, and at most the number of worker threads minus one?

Comment (RandomShaper) [reply doc/classes/ProjectSettings.xml:2702]: It's regardless the setting about system threads. You can think of this in a more abstract way: slots. You have N slots for concurrent tasks. The amount of low-prio slots is computed as `CLAMP(N * low_priority_thread_ratio, 1, cpu_count - 1)`. (N is the amount of CPU cores present.)

For high-priority tasks, slots correspond to pre-allocated threads (the whole point of the thread pool). For low-prio, slots are likewise if `use_system_threads_for_low_priority_tasks` is `false`. If `true`, the slots boil down to new threads spawned and ended each time.

Does this help?

Comment (MewPurPur) [reply doc/classes/ProjectSettings.xml:2702]: I understand that part, I assume you mean `N` means the number of worker threads though.

Comment (RandomShaper) [reply doc/classes/ProjectSettings.xml:2702]: That's right.

Comment (MewPurPur) [reply doc/classes/ProjectSettings.xml:2702]: And you're also saying that low-priority uses its own threads, which won't be able to take on high-priority tasks?

Comment (RandomShaper) [reply doc/classes/ProjectSettings.xml]: First, I have to correct myself regarding my previous comment (after getting a refresher from the code):
- When `use_system_threads_for_low_priority_tasks == false`, the amount of low-prio slots (concurrent low-prio tasks) is computed as `CLAMP(N * low_priority_thread_ratio, 1, cpu_count - 1)`, so there are at least one high-prio and one low-prio slot. (N is the amount of CPU cores present.)
- When `use_system_threads_for_low_priority_tasks == true`, the whole N slots are available for high-prio tasks. Each low-prio task has a thread spawned to process it and then the thread dies.

> And you're also saying that low-priority tasks use their own worker threads, which won't be able to take on high-priority tasks?

Now I do. 😃 (Only true when `use_system_threads_for_low_priority_tasks == true`.)

> Or can the threads also take on high-priority tasks if they aren't occupied?

They can't. Either
- `use_system_threads_for_low_priority_tasks == true` and there's a proper low-prio slot available
- or `use_system_threads_for_low_priority_tasks == false` and thus the low-prio task gets its own throwaway thread.

I think I'm accurate now. I wouldn't mind other eyes rafitying my statements, though.

**Comment by Mickeon** [comment]
Conflicts aside, does this PR look good? Can it be approved?

File: doc/classes/ProjectSettings.xml [hunk 3]
```diff
@@ -2699,17 +2699,19 @@
 			[/codeblock]
 		</member>
 		<member name="threading/worker_pool/low_priority_thread_ratio" type="float" setter="" getter="" default="0.3">
+			If [member threading/worker_pool/use_system_threads_for_low_priority_tasks] is [code]false[/code], this value determines the ratio of [WorkerThreadPool]'s threads that will be reserved for low-priority tasks. For example, if 10 threads are available and this value is set to [code]0.3[/code], 3 of the worker threads will be reserved for low-priority tasks. The actual value won't exceed the number of CPU cores minus one, and if possible, at least one worker thread will be dedicated to low-priority tasks.
```
Comment (RandomShaper) [inline doc/classes/ProjectSettings.xml:2702]: This looks good. When you rebase, you'll have to remove the reference to `use_system_threads_for_low_priority_tasks`, which is now gone (and, so to say, hardcoded to `false`).

Comment (MewPurPur) [reply doc/classes/ProjectSettings.xml]: Forgot to mention that this is doned now

**Review by Mickeon — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**